### PR TITLE
Adds send email to owners of linked issues when ticket closed

### PR DESCRIPTION
**Pull request by ashraffouda**
Signed-off-by: Ashraf Fouda <foudaa@codescalers.com>
issue: https://github.com/gig-projects/org_support/issues/18

### CODE REVIEW

File: _github/github_issue/actions.py [hunk 1]
```diff
@@ -103,15 +103,37 @@ def update_from_github(self,service, event):
...
                 return
 
             if action == 'closed':
-                model = service.model.copy()
-                model['open'] = False
-                model['state'] = 'closed'
-                service.model = model
+                service.model['open'] = False
+                service.model['state'] = 'closed'
+                repo_ays = service.getProducers('github_repo')[0]
```
Comment (zaibon) [inline _github/github_issue/actions.py]: need to handle error case if producer doesn't exists.

File: _github/github_issue/actions.py [hunk 2]
```diff
@@ -103,15 +103,37 @@ def update_from_github(self,service, event):
...
+                repo = repo_ays.actions.get_github_repo(repo_ays)
+                client_ays = repo_ays.getProducers('github_client')[0]
+                client = client_ays.actions.getGithubClient(client_ays)
+                bug = repo.getIssue(service.model['number'])
+                mail_service = service.aysrepo.getService("mailclient", 'main')
+                login = mail_service.hrd.getStr('smtp.login')
+                password = mail_service.hrd.getStr('smtp.passwd')
+                server = mail_service.hrd.getStr('smtp.server')
+                port = mail_service.hrd.getStr('smtp.port')
+                email_sender = j.tools.email.getSender(login, password, server, port)
```
Comment (zaibon) [inline _github/github_issue/actions.py]: maybe mailclient service can have a `getSender` method or something, so it's easy for other services to use.

File: _github/github_issue/actions.py [hunk 3]
```diff
@@ -103,15 +103,37 @@ def update_from_github(self,service, event):
...
+                comments = bug.comments
+                for comment in comments:
+                    issue_num = comment['body'].strip()
```
Comment (zaibon) [inline _github/github_issue/actions.py]: need more logic when parsing comments.

File: _github/github_issue/actions.py [hunk 4]
```diff
@@ -103,15 +103,37 @@ def update_from_github(self,service, event):
...
+                    if issue_num[1:].isnumeric():
+                        issue = repo.getIssue(int(issue_num[1:]))
+                        if "type_ticket" in issue.labels:
+                            user = client.api.get_user(issue.assignee)
+                            email = user.email
+                            email_sender.send(email,
+                                              "support@org_support.com",
```
Comment (zaibon) [inline _github/github_issue/actions.py]: content of email should include solution description of the bug

File: _github/github_issue/actions.py (hunk 1 above)
Comment (rkhamis) [inline _github/github_issue/actions.py]: It will always have its parent as a producer. It's in its schema.

File: _github/github_issue/actions.py [hunk 5]
```diff
@@ -103,15 +103,43 @@ def update_from_github(self,service, event):
...
+                    if repo.type != "code":
+                        return
+                    clients_ays = repo_ays.getProducers('github_client')
+                    if clients_ays:
+                        client_ays = clients_ays[0]
+                        client = client_ays.actions.getGithubClient(client_ays)
+                        bug = repo.getIssue(service.model['number'])
+                        if "type_bug" not in bug.labels:
+                            return
+                        mail_service = service.aysrepo.getService("mailclient", 'main')
```
Comment (zaibon) [inline _github/github_issue/actions.py]: Here you can't be sure that service `mailclient` will exist with the instance name `main`
Or you have to search for a service and handle the case if there is none. Or maybe better github_issue should be able to consume `mailclient`.

File: _github/github_repo/actions.py [hunk 6]
```diff
@@ -114,6 +114,28 @@ def get_issues_from_ays(self, service, refresh=False):
 
         return repo
 
+    def get_github_repo(self,service):
+        githubclientays=service.getProducers('github_client')[0]
+        client = githubclientays.actions.getGithubClient(service=githubclientays)
+        repokey = service.hrd.get("repo.account") + "/" + service.hrd.get("repo.name")
+        repo = client.getRepo(repokey)
+        fromAys = True
+        if service.state.get("getIssuesFromGithub")[0] != "OK":
```
Comment (zaibon) [inline _github/github_repo/actions.py]: `getIssuesFromGithub` can be not present in `service.state`, need to make protect for this case.

File: _github/github_repo/actions.py [hunk 7]
```diff
@@ -114,6 +114,28 @@ def get_issues_from_ays(self, service, refresh=False):
...
+            # means have not been able to get the issues from github properly, so do again
+            fromAys = False
+        if not repo.issues_loaded:
+            if fromAys:
+                print("LOAD ISSUES FROM AYS")
```
Comment (zaibon) [inline _github/github_repo/actions.py]: please use `service.logger` instead of print

File: _github/github_repo/actions.py [hunk 8]
```diff
@@ -114,6 +114,28 @@ def get_issues_from_ays(self, service, refresh=False):
...
+                # service.state.set("getIssuesFromAYS","DO")
+                self.getIssuesFromAYS()
+                repo.issues_loaded = True
+            else:
+                print("LOAD ISSUES FROM GITHUB")
```
Comment (zaibon) [inline _github/github_repo/actions.py]: please use `service.logger` instead of print

**Pull request merged**